### DISCUSSION TITLE
Need a Video Node

### DISCUSSION THREAD

**Discussion by tapaswi-vs** [sceneview/sceneview#503] [Ideas]
I want to render a video when an image is detected. So, far I have used AugmentedImageNode & can attach ModelNode to it but could not find a way to attach a Video to AugmentedImageNode. Is there a VideoNode or something similar so that I can attach a video to an image?

**Comment by jocmp** [comment]
I'm also tracking any progress on the video feature.

There is a commented out `VideoNode` component (https://github.com/SceneView/sceneview-android/commit/d27074f03ee69d68f5c1a3fa02092927bec89780). Based on the discussion in https://github.com/SceneView/sceneview-android/issues/144#issuecomment-2079869137 it stopped working with an earlier version of SceneView and there's no current capacity to enable it.

**Comment by slobodanmilinkovic** [comment]
Any updates on this? I also need VideoNode. It is crucial for my app. As  workaround, I just use the old version 😅

**Comment by ThomasGorisse** [comment]
VideoNode is available in v4.0.0-rc.1:

```kotlin
VideoNode(
    materialLoader = materialLoader,
    mediaPlayer = mediaPlayer,
    width = 2.0f,
    height = 1.5f
)
```

It renders video frames as an `ExternalTexture` on a quad. Works in both `SceneView` and `ARSceneView`.
